currency/pairs_test.go

```diff
@@ -215,3 +239,80 @@ func TestContains(t *testing.T) {
 		t.Errorf("TestContains: Non-existent pair was found")
 	}
 }
+
+func TestDeriveFrom(t *testing.T) {
+	t.Parallel()
+	_, err := Pairs{}.DeriveFrom("")
+	if !errors.Is(err, errPairsEmpty) {
+		t.Fatalf("received: '%v' but expected: '%v'", err, errPairsEmpty)
+	}
+	var testCases = Pairs{
+		NewPair(BTC, USDT),
+		NewPair(USDC, USDT),
+		NewPair(USDC, USD),
+		NewPair(BTC, LTC),
+		NewPair(LTC, SAFEMARS),
+	}
+
+	_, err = testCases.DeriveFrom("")
+	if !errors.Is(err, errSymbolEmpty) {
+		t.Fatalf("received: '%v' but expected: '%v'", err, errSymbolEmpty)
+	}
+
+	_, err = testCases.DeriveFrom("btcUSD")
+	if !errors.Is(err, ErrPairNotFound) {
+		t.Fatalf("received: '%v' but expected: '%v'", err, ErrPairNotFound)
+	}
+
+	got, err := testCases.DeriveFrom("USDCUSD")
+	if !errors.Is(err, nil) {
+		t.Fatalf("received: '%v' but expected: '%v'", err, nil)
+	}
+
+	if got.Upper().String() != "USDCUSD" {
```

Comment:
Just because we experienced it earlier, please ensure that the base and quote are appropriately set so its not USD - CUSD